Content.Server/Interaction/InteractionPopupSystem.cs

```diff
@@ -0,0 +1,63 @@
+using Content.Server.Popups;
+using Content.Server.Interaction.Components;
+using Content.Shared.Interaction;
+using Content.Shared.MobState.Components;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+using Robust.Shared.Random;
+
+
+namespace Content.Server.Interaction;
+
+public sealed class InteractionPopupSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IEntityManager _entities = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<InteractionPopupComponent, InteractHandEvent>(OnInteractHand);
+    }
+
+    private void OnInteractHand(EntityUid uid, InteractionPopupComponent component, InteractHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (_gameTiming.CurTime < component.LastInteractTime + component.InteractDelay)
```

Comment:
Cache CurTime in a local variable because it has to be calculated in the property.